src/main/java/gamelogic/EconoCraftLogic.java

```diff
@@ -84,22 +93,30 @@ public void startEcono() {
 
         while (!exitFlag) {
             ResponseManager.printCurrentRound(playerProfile.getCurrentRound(),
-                    playerProfile.actionPerRound() - actionCount);
+                playerProfile.actionPerRound() - actionCount);
             try {
                 Command command = CommandFactory.create(userInput.nextLine());
                 command.execute(playerProfile);
-                Saver.saveProfile(playerProfile);
+
+                try {
+                    Saver.saveProfile(playerProfile);
+                } catch (SaveProfileException e) {
+                    ResponseManager.indentPrint("Error saving profile: " + e.getMessage());
+                }
+
```

Comment:
Perhaps for this one can try to catch together with CommandInputException and GameException to save space and enhance readability.